cmd/oras/internal/display/metadata/interface.go

```diff
@@ -143,5 +143,5 @@ type RepoListHandler interface {
 	Renderer
 
 	// OnRepositoryListed is called for each repository that is listed.
-	OnRepositoryListed(repo string) error
+	OnRepositoryListed(fullRepo, subRepo string) error
```

Comment:
This abstraction confuses me. `OnRepositoryListed` should pass the full repo only. The namespace should be passed to `NewRepoListHandler`.

Reply:
Nice catch! Updated.